c_glib/arrow-glib/scalar.cpp

```diff
@@ -385,7 +385,9 @@ garrow_scalar_cast(GArrowScalar *scalar,
 {
   const auto arrow_scalar = garrow_scalar_get_raw(scalar);
   const auto arrow_data_type = garrow_data_type_get_raw(data_type);
+  ARROW_SUPPRESS_DEPRECATION_WARNING
   auto arrow_casted_scalar_result = arrow_scalar->CastTo(arrow_data_type);
+  ARROW_UNSUPPRESS_DEPRECATION_WARNING
```

Comment:
Could you use the following instead of this?

```diff
diff --git a/c_glib/arrow-glib/scalar.cpp b/c_glib/arrow-glib/scalar.cpp
index d25bdaf89d..64503ea430 100644
--- a/c_glib/arrow-glib/scalar.cpp
+++ b/c_glib/arrow-glib/scalar.cpp
@@ -26,6 +26,8 @@
 #include <arrow-glib/interval.hpp>
 #include <arrow-glib/scalar.hpp>
 
+#include <arrow/compute/cast.h>
+
 G_BEGIN_DECLS
 
 /**
@@ -385,9 +387,10 @@ garrow_scalar_cast(GArrowScalar *scalar,
 {
   const auto arrow_scalar = garrow_scalar_get_raw(scalar);
   const auto arrow_data_type = garrow_data_type_get_raw(data_type);
-  auto arrow_casted_scalar_result = arrow_scalar->CastTo(arrow_data_type);
+  auto arrow_casted_scalar_result =
+    arrow::compute::Cast(arrow_scalar, arrow_data_type);
   if (garrow::check(error, arrow_casted_scalar_result, "[scalar][cast]")) {
-    auto arrow_casted_scalar = *arrow_casted_scalar_result;
+    auto arrow_casted_scalar = (*arrow_casted_scalar_result).scalar();
     return garrow_scalar_new_raw(&arrow_casted_scalar,
                                  "scalar", &arrow_casted_scalar,
                                  "data-type", data_type,
```

Reply:
(A separated PR is better.)

Reply:
@kou hello. 

Is it recommended to create a new issue for making a new PR, or is it okay if multiple PRs are linked to an existing issue?

Reply:
Could you open one issue per PR?